distributed/scheduler.py

```diff
@@ -758,12 +758,13 @@ def remove_worker(self, stream=None, address=None, safe=False):
             del self.worker_bytes[address]
 
             for key in self.has_what.pop(address):
-                self.who_has[key].remove(address)
-                if not self.who_has[key]:
-                    if key in self.tasks:
-                        recommendations[key] = 'released'
-                    else:
-                        recommendations[key] = 'forgotten'
+                if key in self.who_has:
```

Comment:
`distributed/tests/test_client.py::test_scatter_hash` and `distributed/tests/test_client.py::test__scatter_types`.